pkg/operator/utils/shared_state.go

```diff
@@ -0,0 +1,47 @@
+package utils
+
+import "sync"
+
+type OperatorSharedState struct {
+	defaultDatastore    string
+	csiConfigMapCreated bool
+	sync.RWMutex
+}
+
+var globalSharedState *OperatorSharedState
+
+var initStateSync sync.Once
+
+func InitGlobalState() {
+	initStateSync.Do(func() {
+		globalSharedState = &OperatorSharedState{}
+	})
+}
+
+func GetGlobalSharedState() *OperatorSharedState {
+	return globalSharedState
+}
+
+func (g *OperatorSharedState) GetDefaultDatastore() string {
+	g.RLock()
+	defer g.RUnlock()
+	return g.defaultDatastore
+}
+
+func (g *OperatorSharedState) GetCSIConfigState() bool {
+	g.RLock()
+	defer g.RUnlock()
+	return g.csiConfigMapCreated
+}
+
+func (g *OperatorSharedState) SetCSIConfigState(state bool) {
+	g.Lock()
+	defer g.Unlock()
+	g.csiConfigMapCreated = state
+}
+
+func (g *OperatorSharedState) SetDefaultDatastore(datastorURL string) {
+	g.Lock()
+	defer g.Unlock()
+	g.defaultDatastore = datastorURL
+}
```

Comment:
I'm not sure if I understand why we need a global state. Typically a ConfigMap is used for that, but IIUC you're replacing it with a global object.

IMO it's a bit more complicated this way. Perhaps it's also error-prone, since the same information is stored in 2 different places?

Reply:
Removed this particular field. The shared state is currently used for signalling to target_config controller that, the configmap has been created, so as it can remove stale conditions.